lib/MusicBrainz/Server/Entity/ArtistCredit.pm

```diff
@@ -35,10 +36,15 @@ has 'artist_count' => (
 
 sub is_equal {
     my ($a, $b) = @_;
+
     return 0 unless
         (defined($a) && defined($b)) &&
-        (ref($a) eq ref($b)) &&
-        ($a->name_count == $b->name_count);
+        (ref($a) eq ref($b));
+
+    # refaddr is needed since == is overloaded
+    return 1 if refaddr($a) == refaddr($b);
```

Comment:
How does it differ from the above `(ref($a) eq ref($b))`?

Reply:
`ref` returns a string with the type of reference or name of the class (so `'MusicBrainz::Server::Entity::ArtistCredit'`, for example). That's checking that they're both artist credits. https://perldoc.perl.org/functions/ref.html

`refaddr` returns the memory address, a number. That line checks if they're the same instance in memory. Normally `==` would accomplish that, but it's overloaded to point to `is_equal`, so it'd cause infinite recursion. https://metacpan.org/pod/Scalar::Util#refaddr